### PR TITLE
Convert tests to typescript

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+const mocha = require('gulp-spawn-mocha');
 
-gulp.task('test', () =>
-  gulp.src('test/*_test.js', {read: false})
-      .pipe(mocha({
-        ui: 'tdd',
-        reporter: 'spec',
-      }))
-);
+gulp.task('test', function() {
```
Comment (FredKSchott) [inline gulpfile.js]: I don't love gulp + mocha, the need for gulp-spawn-mocha, and the unnecessary need for istanbul. If we're not doing anything special here, can we just run this via an npm build script? We're already doing this in `.vscode/`

Comment (rictic) [inline gulpfile.js]: Speaking my language. Done.

File: src/test/make_app_test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (justinfagnani) [inline src/test/make_app_test.ts]: remove

File: src/test/cli_test.ts [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
  * Code distributed by Google as part of the polymer project is also
- * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
  */
 
 'use strict';
```
Comment (justinfagnani) [inline src/test/cli_test.ts]: remove

File: src/test/bower_config_test.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (justinfagnani) [inline src/test/bower_config_test.ts]: remove

File: src/test/start_server_test.ts [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import {getApp, ServerOptions} from '../start_server';
```
Comment (justinfagnani) [inline src/test/start_server_test.ts]: Did clang order this way? I think our import order would put paths below names

**Comment by rictic** [comment]
merged into master

**Pull request closed without merging**